### DISCUSSION TITLE
How to use more type when I use web socket?

### DISCUSSION THREAD

**Discussion by wangqiim** [foxglove/community#774] [General]
In [ws-protocol](https://github.com/foxglove/ws-protocol/tree/main) cpp example program, there are some proto I can use. But if I want to use other types, for example [marker](http://docs.ros.org/en/api/visualization_msgs/html/msg/Marker.html)(foxlove studio can display it), whether there is a Marker proto msg in foxglove ws-protocol, or other way i can do this?

**Comment by defunctzombie** [comment]
Marker is a ROS message and ROS does not use protobuf. If you want to use a ROS message you would do so from a ROS system.

**Comment by jtbandes** [comment]
We publish [protos](https://github.com/foxglove/schemas/tree/main/schemas/proto/foxglove) for the [Foxglove schemas](https://foxglove.dev/docs/studio/messages/introduction). The equivalent of a ROS Marker would be a [foxglove.SceneUpdate](https://foxglove.dev/docs/studio/messages/scene-update).

You can also embed ROS data as long as you use the correct encodings as documented here: https://foxglove.dev/docs/studio/connection/custom#ros-1-and-ros-2

**Reply by wangqiim** [reply]
Thanks, It works!
